spring-messaging/src/jmh/java/org/springframework/messaging/simp/broker/DefaultSubscriptionRegistryBenchmark.java

```diff
@@ -0,0 +1,206 @@
+			for (int subscription = 0; subscription < subscriptionCount; subscription++) {
+				String destination = getDestinationForSubscription(subscription);
+				this.registry.registerSubscription(subscribeMessage(sessionId, "subscription" + subscription, destination));
+			}
+		}
+
+		public String getDestinationForSubscription(int subscription) {
+			return this.destinations[subscription % this.destinations.length];
+		}
+	}
+
+	@State(Scope.Thread)
+	public static class Requests {
+		public static final AtomicInteger counter = new AtomicInteger();
+
+		@Param({"false", "true"})
+		boolean sameDestination;
+
+		@Param({"false", "true"})
+		boolean sameSession;
+
+		public String session;
+
+		public Message<?> subscribe;
+
+		public String findDestination;
+
+		public Message<?> unsubscribe;
+
+		@Setup(Level.Trial)
+		public void doSetup(ServerState serverState) {
+			int someInteger = counter.getAndIncrement();
+			if (this.sameSession) {
+				this.session = serverState.sessionIds[0];
+			}
+			else {
+				this.session = serverState.sessionIds[someInteger % serverState.sessionIds.length];
+			}
+			String subscription = String.valueOf(someInteger);
+			String destination;
+			if (this.sameDestination) {
+				destination = serverState.destinations[0];
+				this.findDestination = serverState.destinations[0];
+			}
+			else {
+				destination = serverState.destinations[someInteger % serverState.destinations.length];
+				this.findDestination = serverState.destinations[someInteger % serverState.destinations.length];
+			}
+			this.subscribe = subscribeMessage(this.session, subscription, destination);
+			this.unsubscribe = unsubscribeMessage(this.session, subscription);
+		}
+
+	}
+
+	@State(Scope.Thread)
+	public static class UnregisterAllSubscriptionRequest {
+
+		@Param("20")
+		public int subscriptionCount;
+
+		public String session;
+
+		@Setup(Level.Invocation)
+		public void doSetup(ServerState serverState) {
+			this.session = UUID.randomUUID().toString();
+			serverState.registerSubscriptions(this.session, this.subscriptionCount);
+		}
+	}
+
+	@State(Scope.Thread)
+	public static class FindRequest {
+		public static final AtomicInteger counter = new AtomicInteger();
+
+		public String destination;
+
+		public Message<?> message;
+
+		@Setup(Level.Trial)
+		public void doSetup(ServerState serverState) {
+			this.destination = serverState.destinations[this.counter.getAndIncrement() % serverState.destinations.length];
+			this.message = null;
+		}
+	}
+
+	@Benchmark
+	public void registerSubscription(ServerState serverState, Requests request) {
+		serverState.registry.registerSubscription(request.subscribe);
+		serverState.registry.findSubscriptionsInternal(request.findDestination, null);
+		serverState.registry.unregisterSubscription(request.unsubscribe);
+	}
+
+	@Benchmark
+	public void unregisterSession(ServerState serverState, UnregisterAllSubscriptionRequest request) {
```

Comment:
without a way to consume the result of the operation (maybe consuming the list of subscriptions after unregistering?), maybe removing the benchmark is the best course of action...

Reply:
Your are right, the JVM eliminated that code and when I add a findSubscriptionsInternal() the "performance" dropped. Not it looks like it is measuring something meaningful. Good finding, thank you.